tests/CMakeLists.txt

```diff
@@ -12,48 +12,77 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-add_subdirectory( ArrayProxy )
-add_subdirectory( ArrayProxyNoTemporaries )
-add_subdirectory( ArrayWrapper )
-if( VULKAN_HPP_ENABLE_CPP20_MODULES )
-	add_subdirectory( Cpp20Modules )
-	if( VULKAN_HPP_ENABLE_STD_MODULE )
-		add_subdirectory( CppStdModule )
-	endif()
-endif()
-add_subdirectory( CppType )
+vulkan_hpp__setup_test( NAME ArrayProxy )
+vulkan_hpp__setup_test( NAME ArrayProxyNoTemporaries )
+vulkan_hpp__setup_test( NAME ArrayWrapper )
+vulkan_hpp__setup_test( NAME CppType )
 if( CMAKE_CXX_STANDARD GREATER_EQUAL 20 )
-	add_subdirectory( DesignatedInitializers )
+	vulkan_hpp__setup_test( NAME DesignatedInitializers )
+	vulkan_hpp__setup_test( NAME ExtensionInspection )
 endif()
-add_subdirectory( DeviceFunctions )
-add_subdirectory( DispatchLoaderDynamic )
-add_subdirectory( DispatchLoaderDynamicSharedLibrary )
-add_subdirectory( DispatchLoaderDynamicSharedLibraryClient )
-add_subdirectory( DispatchLoaderStatic )
-add_subdirectory( EnableBetaExtensions )
-if( CMAKE_CXX_STANDARD GREATER_EQUAL 20 )
-	add_subdirectory( ExtensionInspection )
+# without ctest, can only run these tests with VULKAN_HPP_DISPATCH_LOADER_DYNAMIC disabled
+# CI cannot run these tests either way, as it does not have access to the Vulkan package
+if( NOT VULKAN_HPP_DISPATCH_LOADER_DYNAMIC )
+	vulkan_hpp__setup_test( NAME DeviceFunctions )
+	vulkan_hpp__setup_test( NAME DispatchLoaderStatic )
+endif()
+# without ctest, can only run these tests with VULKAN_HPP_DISPATCH_LOADER_DYNAMIC enabled
+if( NOT( NOT VULKAN_HPP_TESTS_CTEST AND NOT VULKAN_HPP_DISPATCH_LOADER_DYNAMIC ))
```

Comment:
Isn't this check equivalent to
`if( VULKAN_HPP_TESTS_CTEST OR VULKAN_HPP_DISPATCH_LOADER_DYNAMIC)`